engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -76,6 +76,33 @@
   private String recordingNameTopic;
   private long numWindows = Long.MAX_VALUE; // number of windows to record
   private Runnable stopProcedure; // stop procedure to execute
+
+  private static final Logger logger = LoggerFactory.getLogger(TupleRecorder.class);
+
+  // If there are errors processing tuples, don't log an error for every tuple as it could overwhelm the logs.
+  // The property specifies the minumum number of tuples between two consecutive error log statements. Set it to zero to
+  // log every tuple error
+  private static int ERROR_LOG_GAP;
+  int lastLog = -1;
+
+  static {
+    ERROR_LOG_GAP = 10000;
+    String property = System.getProperty("apex.tupleRecorder.errorLogGap");
+    if (property != null) {
+      try {
+        int value = Integer.parseInt(property);
```

Comment:
I prefer to let the user know what the problem is, so do not prefer getInteger(). decode() seems to be able to handle more formats and fine with it, though not sure if we need that level of flexibility for user to be able to specify octal, hexadecimal formats etc..